mala/network/runner.py

```diff
@@ -41,6 +49,334 @@ def __init__(self, params, network, data, runner_dict=None):
         self.data = data
         self.__prepare_to_run()
 
+    def _calculate_errors(
+        self, actual_outputs, predicted_outputs, metrics, snapshot_number
+    ):
+        """
+        Calculate the errors between the actual and predicted outputs.
+
+        Parameters
+        ----------
+        actual_outputs : numpy.ndarray
+            Actual outputs.
+
+        predicted_outputs : numpy.ndarray
+            Predicted outputs.
+
+        metrics : list
+            List of metrics to calculate.
+
+        snapshot_number : int
+            Snapshot number for which the errors are calculated.
+
+        Returns
+        -------
+        errors : dict
+            Dictionary containing the errors.
+        """
+        energy_metrics = [metric for metric in metrics if "energy" in metric]
```

Comment:
There is one metric which is not reflected in this new workflow, but which I think maybe should, that's the "_energy_full" metrics I implemented for band and total energy. The idea there was to give users the option to return either both the actual band/total energy using the Tester class. For some visualizations (e.g. the correlation plot, Fig. 4 of https://www.nature.com/articles/s41524-023-01070-z.pdf) it is nice to have the energies of all test snapshots (both DFT, actual and predicted) as one big table. 

Since the changes here unify Runner, Trainer and Tester, we would probably have to safeguard this option to only be accessible from the Tester class.

Reply:
I am not a big fan of functions which change their return type (here: number vs list with two numbers) based on the arguments. I'll add a separate function which provides this functionality.

Reply:
It's now implemented in Tester.get_energy_targets_and_predictions()
Let me know if you think it should be done differently.

Reply:
Good point, I agree that having a consistent return type per function is generally a good idea. I like the idea with the extra function, thank you!